File: commands/tokens.js
const { tokenManager } = require("../tokenManager");

module.exports = {
  run: async ({ username, message }) => {
    try {
      const normalizedUsername = username.toLowerCase();
      
      const args = message.split(' ');
      let targetUser = normalizedUsername;
      let displayTargetUser = username;
      
      if (args.length > 1 && args[1].startsWith('@')) {
        targetUser = args[1].substring(1).toLowerCase();
        displayTargetUser = args[1].substring(1);
      } else if (args.length > 1) {
        targetUser = args[1].toLowerCase();
        displayTargetUser = args[1];
      }

      const userData = await tokenManager.getUser(targetUser);
      
      if (targetUser === normalizedUsername) {
        await global.sendChatMessage(
          `@${username}: You have ${userData.tokens} tokens! Total earned: ${userData.totalEarned}, Total spent: ${userData.totalSpent}`
        );
      } else {
        if (userData.totalEarned > 0 || userData.totalSpent > 0) {
          await global.sendChatMessage(
            `@${username}: ${displayTargetUser} has ${userData.tokens} tokens (earned: ${userData.totalEarned}, spent: ${userData.totalSpent})`
          );
        } else {
          await global.sendChatMessage(
            `@${username}: ${displayTargetUser} has no token activity yet`
          );
        }
      }

    } catch (error) {
      console.error("Error in tokens command:", error);
      await global.sendChatMessage(`@${username}: Error checking tokens`);
    }
  }
};

File: KickBot/Commands/Tokens.cs
namespace KickBot.Commands;

public class TokensCommand : ICommand
{
    public string Name => "tokens";
    public string Description => "Check your token balance or another user's balance";
    public bool BroadcasterOnly => false;

    public async Task ExecuteAsync(string username, string[] args, CommandContext context)
    {
        var normalizedUsername = username.ToLower();
        string targetUser = normalizedUsername;
        string displayTargetUser = username;

        if (args.Length > 0)
        {
            targetUser = args[0].TrimStart('@').ToLower();
            displayTargetUser = args[0].TrimStart('@');
        }

        var userData = context.UserManager.GetOrCreateUser(targetUser);

        if (targetUser == normalizedUsername)
        {
            await context.SendMessage(
                context.BroadcasterUserId,
                $"@{username}: You have {userData.Tokens} tokens! Total earned: {userData.TotalEarned}, Total spent: {userData.TotalSpent}"
            );
        }
        else
        {
            if (userData.TotalEarned > 0 || userData.TotalSpent > 0)
            {
                await context.SendMessage(
                    context.BroadcasterUserId,
                    $"@{username}: {displayTargetUser} has {userData.Tokens} tokens (earned: {userData.TotalEarned}, spent: {userData.TotalSpent})"
                );
            }
            else
            {
                await context.SendMessage(
                    context.BroadcasterUserId,
                    $"@{username}: {displayTargetUser} has no token activity yet"
                );
            }
        }

        Console.WriteLine($"[COMMAND] {username} checked tokens for {displayTargetUser}");
    }
}